sources/web/datalab/polymer/components/datalab-files/datalab-files.ts

```diff
@@ -392,90 +392,87 @@ class FilesElement extends Polymer.Element {
+        okLabel: 'Upload Anyway',
+      };
 
-      let proceedWithUpload = true;
-      // If there's at least one large file, show a dialog to confirm the user
-      // wants to continue with the upload.
-      if (hasLargeFile) {
-        let warningMsg = files.length > 1 ? 'Some of the files you selected are '
-                                            : 'The file you selected is ';
-        warningMsg += 'larger than 25MB. You might experience browser freeze or crash';
-        const dialogOptions: DialogOptions = {
-          title: 'Warning: Large File',
-          messageHtml: warningMsg,
-          okLabel: 'Upload Anyway',
-        };
-
-        const result: BaseDialogCloseResult = 
-          await Utils.showDialog(BaseDialogElement, dialogOptions);
-          proceedWithUpload = result.confirmed === true;
-      }
+      const result: BaseDialogCloseResult = 
+        await Utils.showDialog(BaseDialogElement, dialogOptions);
+        proceedWithUpload = result.confirmed === true;
+    }
 
-      if (proceedWithUpload) {
-        for (let i = 0; i < files.length; ++i) {
-          const file = files[i];
-
-          // First, load the file data into memory.
-          const loadPromise = new Promise((resolve, reject) => {
-
-            const reader = new FileReader();
-
-            reader.onload = function() {
-              let itemData = reader.result;
-              // Extract the base64 data string
-              itemData = itemData.substr(itemData.indexOf(',') + 1);
-
-              const model: JupyterFile = {
-                name: file.name,
-                path: currentPath,
-                type: 'file',
-                format: 'base64',
-                content: itemData,
-              };
-
-              resolve(model);
-            }
-
-            // TODO: handle file reading errors.
-            reader.onerror = () => {
-              reject('Error reading file.');
-            }
-
-            // TODO: this will freeze the UI on large files (>~20MB on my laptop) until
-            // they're loaded into memory, and very large files (>~100MB) will crash
-            // the browser.
-            // One possible solution is to slice the file into small chunks and upload
-            // each separately, but this requires the backend to support partial
-            // chunk uploads. For Jupyter, this is supported in 5.0.0, see:
-            // https://github.com/jupyter/notebook/pull/2162/files
-            reader.readAsDataURL(file);
-          });
-
-          // Now upload the file data to the backend server.
-          const uploadPromise = loadPromise
-            .then((model: JupyterFile) => ApiManager.saveJupyterFile(model));
-          uploadPromises.push(uploadPromise);
-        }
-      }
+    if (proceedWithUpload) {
+      files.forEach((file: File) => {
+
+        // First, load the file data into memory.
+        const loadPromise = new Promise((resolve, reject) => {
```

Comment:
Nit: promise creation should be as narrowly scoped as possible to make it easier to validate that it is always resolved.

Would recommend just:

```
const loadPromise = new Promise((resolve, reject) => {
  const reader = new FileReader();
  reader.onload = () => reader.result;
  reader.onerror = (e) => throw new Error('Error reading file');
  reader.readAsDataURL(file);
});
```